packages/server-wallet/src/protocols/process-ledger-queue.ts

```diff
@@ -0,0 +1,198 @@
+import {compose, map, filter} from 'lodash/fp';
+import _ from 'lodash';
+import {
+  allocateToTarget,
+  AllocationItem,
+  BN,
+  checkThat,
+  isSimpleAllocation,
+  Outcome,
+  SignedStateWithHash,
+  SimpleAllocation,
+} from '@statechannels/wallet-core';
+import {Transaction} from 'knex';
+
+import {Store} from '../wallet/store';
+import {Bytes32} from '../type-aliases';
+import {LedgerRequestType} from '../models/ledger-requests';
+
+import {Protocol, ProtocolResult, ChannelState} from './state';
+import {
+  MarkLedgerFundingRequestsAsComplete,
+  noAction,
+  ProtocolAction,
+  SignState,
+  signState,
+} from './actions';
+
+export type ProtocolState = {
+  fundingChannel: ChannelState;
+  channelsPendingRequest: ChannelState[];
+};
+
+const allocationItemComparator = (a: AllocationItem, b: AllocationItem): boolean =>
+  a.destination === b.destination && BN.eq(a.amount, b.amount);
```

Comment:
Could you choose a better name? `areAllocationItemsEqual`? You might reuse some functions from `wallet-core`, or even insert this function into `wallet-core`.

ref:
https://github.com/statechannels/statechannels/blob/98faef14ad8bdf103debcd869f238da13a6a4152/packages/wallet-core/src/state-utils.ts#L104

Reply:
8ce4e2ecd